pkg/apis/config/v1alpha4/types.go

```diff
@@ -142,6 +142,9 @@ type Node struct {
 	// The node-level patches will be applied after the cluster-level patches
 	// have been applied. (See Cluster.KubeadmConfigPatchesJSON6902)
 	KubeadmConfigPatchesJSON6902 []PatchJSON6902 `yaml:"kubeadmConfigPatchesJSON6902,omitempty"`
+
+	// ExtraArguments to be given to the provider
+	ExtraArguments []string `yaml:"extraArguments,omitempty"`
```

Comment:
I'm not in favor of supporting this, the fact that we use exec is an implementation detail.

Reply:
> I'm not in favor of supporting this, the fact that we use exec is an implementation detail.

yeah, but is the only way, no? 
you can not do arbitrary patching ala kubeadm with the providers

Reply:
you don't need to, we can come up with a good way to enable GPUs.

I think the problem is: kubernetes has a pretty good model where you just request N GPUs and GPUs are uniform within a node. One option would be to only support passing through all GPUs to *one* node and just have an option for this.

Reply:
dear @BenTheElder - before I start working on a proposal - can you please confirm that I understood your concerns correctly? is your concern that passing the --gpus option is specific to docker and doesn't work on portman or that you can't control the amount of GPUs assigned to the kind node (e.g. other applications or other kind clusters won't get access to them anymore)